### ISSUE TITLE
Problems with EditorResourcePicker `handle_menu_selected` function signature

### ISSUE THREAD

**Issue by eh-jogos** [godotengine/godot#62679]
### Godot version

3.3.4.stable

### System information

Manjaro Linux, Nvidia proprietary drivers, GLES3

### Issue description

Ok, so I was finally able to use `EditorResourcePicker` to create a plugin for my projects where I can export Custom Resources, but there were a lot of small issues in the way.

The most glaring one, the signature for `handle_menu_selected` is wrong. According to documentation and to the editor's own type completion it should be:
```
func handle_menu_selected(id: int) -> void:
```
But actually when you implement it like that, your code works but gives random errors:
![image](https://user-images.githubusercontent.com/13070158/177044313-5b24ed62-86e3-454d-89ee-00eafef99991.png)

I tried to take a look at the cpp code, I can't write cpp well but I got to the point of at least reading it, and above the error line I find [this](https://github.com/godotengine/godot/blob/2c88a947c1395fb3c8952f2184b83c74c3532744/editor/editor_resource_picker.cpp#L340):
![image](https://user-images.githubusercontent.com/13070158/177044428-ed0e3896-22c0-4c36-9748-5eae32eccefc.png)

And I was like "huh? I though this was a void function", so I Ctrl+F this function and find out it is indeed a `bool` [function](https://github.com/godotengine/godot/blob/2c88a947c1395fb3c8952f2184b83c74c3532744/editor/editor_resource_picker.cpp#L446), but if I try to implement it as `bool` on gdscript I get an error:
![image](https://user-images.githubusercontent.com/13070158/177044553-8ddb7488-7663-4967-ba22-e62a4a0eb4fc.png)
So I actually have to remove the type hints and let duck typing do it's magic and it works, I stop getting the error messages.

So I guess somewhere in cpp this is being exposed to GDScript as a void? And that must be making the documentation wrong as well?

There are other issues with `EditorResourcePicker` that are not obvious, for example when implementing `func set_create_options(menu_node: Object) -> void:` you have to give it ids that do not conflict with the built in menu options, but I can't get the size of them because at this point if I try to do:
```
menu_node.get_item_count()
```
It prints 0. But if I start my custom entries with id at 0, they will never call `handle_menu_selected` as from what I observed:
- `id = 0` - open "Load" popup
- `id = 1` - open "Quick Load" popup
- `id = 2` - same as clicking "Edit"
- `id = 3` - same as clicking "Clear"

I didn't have more options to test, and I guess it makes sense that the initial ids are taken, but `EditorResourcePicker` offers no documentation on them, no constant I could use as the first safe id and build from there, or an enum with the taken ids.

Should I open a second issue for this "id" confusion? Or maybe just an issue in the documentation repository? 

### Steps to reproduce

Use the MRP below

### Minimal reproduction project

[editor_resource_picker_problems.zip](https://github.com/godotengine/godot/files/9034996/editor_resource_picker_problems.zip)

The MRP has my "[personal library addon](https://github.com/eh-jogos/eh_Utilities/tree/submodule)" as that's where I made the custom resource inspector addon and it was easier to just bring everything instead of only the plugin, but the relevant files for this problem are all in the folder:
**res://addons/eh_jogos.utilities/custom_inspectors/custom_resource/**

The main file to edit to reproduce the errors is `custom_resource_picker.gd`
I left some comments there of how to reproduce each error:
- **for the function signature error:** 
![image](https://user-images.githubusercontent.com/13070158/177045193-3e397212-2ef5-4d66-882d-f8331d0d59c4.png)
- **for the id count errors:**:
![image](https://user-images.githubusercontent.com/13070158/177045216-60124bcf-7493-42a8-aec9-9199dfef9c3a.png)

`Node2D.gd` use advanced exports to create a property in the editor with a custom hint_string
![image](https://user-images.githubusercontent.com/13070158/177045289-a62b1af2-98f9-432f-b3bb-82e135b32874.png)

And the other files in the **res://addons/eh_jogos.utilities/custom_inspectors/custom_resource/** are to handle that custom hint_string and create the `EditorProperty` that uses the `custom_resource_picker.gd`

So you can just select the root node of the main scene and play with it in the editor:
![image](https://user-images.githubusercontent.com/13070158/177045371-84587d11-f84c-4767-90c2-7d81068d629d.png)

There are also some resources saved on disk on the folder `res://test/` to test the drag and drop to the inspector property

**Comment by YuriSizov** [comment]
This was already fixed in https://github.com/godotengine/godot/pull/56325, it will be fixed in 3.5. I'll mark it for cherry-picking for 3.4 and 3.3, I guess.

**Comment by eh-jogos** [comment]
> This was already fixed in #56325, it will be fixed in 3.5. I'll mark it for cherry-picking for `3.4`, I guess. (You mention that you're on `3.3.4`, but `EditorResourcePicker` was only added in `3.4`).

Sorry, I meant 3.4.4, got the wrong number there.

Thanks for the reply, sorry for the duplicated issue, I forgot to search on closed ones again and only searched for opened issues.

**Comment by YuriSizov** [comment]
No worries, it wasn't a reported issue! I just noticed it while working on code and fixed it 🙃
